lib/astarte_streams/k8s.ex

```diff
@@ -0,0 +1,151 @@
+#
+# This file is part of Astarte.
+#
+# Copyright 2020 Ispirata Srl
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+defmodule Astarte.Streams.K8s do
+  require Logger
+
+  alias K8s.Client
+
+  @api_version "api.astarte-platform.org/v1alpha1"
+  @flow_kind "Flow"
+
+  defmodule Container do
+    @enforce_keys [
+      :block_id,
+      :image,
+      :exchange_routing_key,
+      :queue,
+      :cpu_limit,
+      :memory_limit,
+      :cpu_requests,
```

Comment:
I would not enforce requests since a container could not need them

Reply:
Let's force it right now, we'll make it optional on a different layer.